src/lib.rs

```diff
@@ -732,36 +732,119 @@ impl<'a, R: Rng> Iterator for AsciiGenerator<'a, R> {
+/// A random number generator that can be explicitly seeded.
+///
+/// Each pseudo-random number generator (PRNG) should implement this.
+pub trait SeedableRng: Sized {
+    /// Seed type, which is restricted to `u8` arrays with a length of
+    /// 4, 8, 12, 16, 24 and 32.
     ///
-    /// # Example
+    /// It is recommended to seed PRNG's with a seed of more than circa
+    /// 100 bits, which means an array of `[u8; 12]` or greater to avoid picking
+    /// RNG's with partially overlapping periods.
     ///
-    /// ```rust
-    /// use rand::{Rng, SeedableRng, StdRng};
+    /// For cryptographic RNG's a seed of 256 bits is recommended, `[u8; 32]`.
+    type Seed: Sized + Default + AsMut<[u8]>;
+
+    /// Create a new PRNG using the given seed.
     ///
-    /// let seed: &[_] = &[1, 2, 3, 4];
-    /// let mut rng: StdRng = SeedableRng::from_seed(seed);
-    /// println!("{}", rng.gen::<f64>());
-    /// rng.reseed(&[5, 6, 7, 8]);
-    /// println!("{}", rng.gen::<f64>());
-    /// ```
-    fn reseed(&mut self, Seed);
+    /// PRNG implementations are allowed to assume that bits in the seed are
+    /// well distributed. That means usually that the number of one and zero
+    /// bits are about equal, and values like 0, 1 and (size - 1) are unlikely.
+    ///
+    /// PRNG implementations are recommended to be reproducible. A PRNG seeded
+    /// using this function with a fixed seed should produce the same sequence
+    /// of output in the future and on different architectures (with for example
+    /// different endianness).
+    ///
+    /// It is however not required that this function yield the same state as a
+    /// reference implementation of the PRNG given equivalent seed; if necessary
+    /// another constructor can be used.
+    fn from_seed(seed: Self::Seed) -> Self;
 
-    /// Create a new RNG with the given seed.
+    /// Create a new PRNG seeded from another `Rng`.
     ///
-    /// # Example
+    /// This is the recommended way to initialize PRNGs. See the `NewRng`
+    /// trait that provides a convenient `new` method using `from_rng` and a
+    /// good entropy source.
     ///
-    /// ```rust
-    /// use rand::{Rng, SeedableRng, StdRng};
+    /// It is recommended to use a good source of randomness to initialize the
+    /// PRNG. Otherwise small PRNG's could show statistical bias in the first
+    /// couple of results, and possibly not use their entire period well.
+    /// Cryptographic PRNG's can be less secure or even insecure when they are
+    /// seeded from a non-cryptographic PRNG.
     ///
-    /// let seed: &[_] = &[1, 2, 3, 4];
-    /// let mut rng: StdRng = SeedableRng::from_seed(seed);
-    /// println!("{}", rng.gen::<f64>());
-    /// ```
-    fn from_seed(seed: Seed) -> Self;
+    /// Examples of good RNG's for seeding are entropy sources like `OsRng` and
+    /// `JitterRng`. Also cryptographically secure PRNG's (like `thread_rng`)
+    /// can be used without hesitation.
```

Comment:
Prefer to seed from a strong external entropy source like `OsRng` or from a cryptographic PRNG.